File: PDOO-Civitas/CivitasR/lib/sorpresa.rb
#encoding: utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Sorpresa
   
    public
    def initialize()
      
      @valor = -1
      
    end
    
    public
    def jugador_correcto(actual, todos)
      
      return (actual < todos.length())
      
    end
    
    public
    def salir_del_mazo()
      
      if(@valor == 1)
        
        @mazo.inhabilitar_carta_especial(self)
      end
      
    end
    
    public
    def usada()
      
      if(@valor == 1)
        
        @mazo.habilitar_carta_especial(self)
      end
      
    end

    private_class_method :new
    
  end

end

File: PDOO-Civitas/CivitasJ/src/civitas/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ramonjtorres
 * @author david
 */
abstract class Sorpresa {
    
    private int valor;
    MazoSorpresas mazo;
    String informacion;
    
    public String getInformacion(){
    
        return informacion;
    }
    
    abstract void aplicarAJugador(int actual, ArrayList<Jugador> todos);
    
    abstract void informe(int actual, ArrayList<Jugador> todos);
    
    public boolean jugadorCorrecto(int actual, ArrayList<Jugador> todos){
        
        return (actual < todos.size());
    }
    
    void salirDelMazo(){
    
        if(this.valor == 1){
        
            mazo.inhabilitarCartaEspecial(this);
        }
    }
    
    void usada(){
    
        if(this.valor == 1){
        
            mazo.habilitarCartaEspecial(this);
        }
    }
    
    public abstract String toString();
}